symfony/src/Service/MayolinkOrderService.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace App\Service;
+
+use Pmi\HttpClient;
+use Psr\Log\LoggerInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Twig\Environment;
+use App\Entity\Order;
+
+class MayolinkOrderService
+{
+    protected $params;
+    protected $em;
+    protected $twig;
+    protected $logger;
+    protected $client;
+    protected $ordersEndpoint;
+    // This namespace is the same across all environments, regardless of the endpoint.
+    // Also, note that this is just an XML namespace and is never used to make a request
+    protected $nameSpace = 'http://orders.mayomedicallaboratories.com';
+    protected $labelPdf = 'orders/labels.xml';
+    protected $createOrder = 'orders/create.xml';
+    protected $userName;
+    protected $password;
+
+
+    public function __construct(ParameterBagInterface $params, EntityManagerInterface $em, Environment $twig, LoggerInterface $logger)
+    {
+        $this->params = $params;
+        $this->em = $em;
+        $this->twig = $twig;
+        $this->logger = $logger;
+        $this->client = new HttpClient(['cookies' => true]);
+        $this->ordersEndpoint = $params->get('ml_orders_endpoint');
+        $this->userName = $params->get('ml_username');
+        $this->password = $params->get('ml_password');
+        if (empty($this->ordersEndpoint) || empty($this->userName) || empty($this->password)) {
+            throw new \Exception('MayoLINK connection is not configured.');
+        }
```

Comment:
Previously, we could have `ml_mock_order` defined locally and not need to specify the endpoint and credentials. Can we reinstate this behavior? Currently, this service needs to be wired even if using the mock orders, and it fails when the endpoint and credentials aren't defined.